library/core/src/intrinsics/mod.rs

```diff
@@ -2042,6 +2046,18 @@ pub unsafe fn volatile_copy_memory<T>(_dst: *mut T, _src: *const T, _count: usiz
 /// The volatile parameter is set to `true`, so it will not be optimized out
 /// unless size is equal to zero.
 ///
+/// # Safety
+///
+/// Behavior is undefined if any of the following conditions are violated:
+///
+/// * `_dst` must be [valid] for writes of `_count * size_of::<T>()` bytes.
+///
+/// * `_dst` must be properly aligned.
+///
+/// Note that even if `T` has size `0`, the pointer must be properly aligned.
```

Comment:
You can just point at `write_bytes` here, the safety requirements are the same.

Except of course they aren't really -- these are *volatile* operations which can be used in ways that would be invalid for `write_bytes`. The same applies to `copy_nonoverlapping`  vs `volatile_copy_nonoverlapping_memory`.